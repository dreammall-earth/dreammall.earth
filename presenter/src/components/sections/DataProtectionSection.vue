<template>
  <div class="data-protection-section pt-4 v-container">
    <v-row class="section-header">
      <v-col>
        <h1 class="section-headline">
          {{ $t('dataProtection.headline') }}
        </h1>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-card flat class="content-card mb-2 pa-4 pa-md-12 h-100">
          <template #text>
            <h2 class="card-header">{{ $t('dataProtection.general.headline') }}</h2>
            <p class="card-content my-4">{{ $t('dataProtection.general.infoStart') }}</p>
            <ul class="card-content mt-2">
              <li>{{ $t('dataProtection.general.company') }}</li>
              <li>{{ $t('dataProtection.general.person') }}</li>
              <li>{{ $t('dataProtection.general.street') }}</li>
              <li>
                {{ $t('dataProtection.general.zip') }} {{ $t('dataProtection.general.city') }}
              </li>
            </ul>
            <p class="card-content my-4">{{ $t('dataProtection.general.infoEnd') }}</p>
          </template>
        </v-card>
      </v-col>
      <v-col>
        <v-card flat class="content-card mb-2 pa-4 pa-md-10 h-100">
          <template #text>
            <h2 class="card-header">{{ $t('dataProtection.revocation.headline') }}</h2>
            <p class="card-content my-4">{{ $t('dataProtection.revocation.content') }}</p>
          </template>
        </v-card>
      </v-col>
    </v-row>

    <v-row>
      <v-col>
        <v-card flat class="content-card mb-2 pa-4 pa-md-12">
          <template #text>
            <h2 class="card-header">{{ $t('dataProtection.euDispute.headline') }}</h2>
            <p class="card-content mt-2">
              {{ $t('dataProtection.euDispute.content') }}
            </p>
          </template>
        </v-card>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-card flat class="content-card mb-2 pa-4 pa-md-12">
          <template #text>
            <h2 class="card-header">{{ $t('dataProtection.consumerDispute.headline') }}</h2>
            <p class="card-content mt-2">
              {{ $t('dataProtection.consumerDispute.content') }}
            </p>
          </template>
        </v-card>
      </v-col>
    </v-row>
  </div>
</template>

<script lang="ts" setup></script>

<style scoped lang="scss">
$mobile: 500px;

.data-protection-section {
  max-width: 1120px;
  font-family: Poppins, sans-serif;
  color: #383838;
  background: #f5f5f5;

  .section-header {
    text-align: center;

    h1.section-headline {
      font-size: 3.875rem;
      font-style: normal;
      font-weight: 700;
      line-height: 120%;
      text-transform: capitalize;

      @media screen and (max-width: $mobile) {
        font-size: 2.5rem;
      }
    }
  }

  .content-card {
    font-size: 1.25rem;
    line-height: normal;
    background: #fff;
    border-radius: 1.875rem;

    @media screen and (max-width: $mobile) {
      font-size: 1rem;
    }

    .card-header {
      font-weight: 700;

      @media screen and (max-width: $mobile) {
        font-size: 1rem;
      }
    }

    .card-content {
      font-weight: 300;
      list-style: none;
    }
  }
}
</style>
