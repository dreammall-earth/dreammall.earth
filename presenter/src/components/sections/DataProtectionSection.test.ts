import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach } from 'vitest'

import DataProtectionSection from './DataProtectionSection.vue'

describe('DataProtectionSection', () => {
  const Wrapper = () => {
    return mount(DataProtectionSection)
  }
  let wrapper: ReturnType<typeof Wrapper>

  beforeEach(() => {
    wrapper = Wrapper()
  })

  it('renders 4 cards', () => {
    expect(wrapper.find('.data-protection-section').exists()).toBeTruthy()
    expect(wrapper.findAll('.v-col > .v-card')).toHaveLength(4)
  })

  it('has h1 dataProtection.headline', () => {
    expect(wrapper.find('.data-protection-section h1.section-headline').text()).toBe(
      "$t('dataProtection.headline')",
    )
  })

  describe('first card', () => {
    it('has h2 dataProtection.general.headline', () => {
      expect(wrapper.findAll('.v-col > .v-card')[0].findAll('h2.card-header')[0].text()).toBe(
        "$t('dataProtection.general.headline')",
      )
    })
  })

  describe('second card', () => {
    it('has h2 dataProtection.processing.headline', () => {
      expect(wrapper.findAll('.v-col > .v-card')[1].find('h2.card-header').text()).toBe(
        "$t('dataProtection.processing.headline')",
      )
    })
  })

  describe('third card', () => {
    it('has h2 dataProtection.revocation.headline', () => {
      expect(wrapper.findAll('.v-col > .v-card')[2].find('h2.card-header').text()).toBe(
        "$t('dataProtection.revocation.headline')",
      )
    })

    it('has p dataProtection.revocation.content', () => {
      expect(wrapper.findAll('.v-col > .v-card')[2].find('p.card-content').text()).toBe(
        "$t('dataProtection.revocation.content')",
      )
    })
  })

  describe('fourth card', () => {
    it('has h2 dataProtection.euDispute.headline', () => {
      expect(wrapper.findAll('.v-col > .v-card')[3].find('h2.card-header').text()).toBe(
        "$t('dataProtection.euDispute.headline')",
      )
    })

    it('has p dataProtection.euDispute.content', () => {
      expect(wrapper.findAll('.v-col > .v-card')[3].find('p.card-content').text()).toBe(
        "$t('dataProtection.euDispute.content')",
      )
    })
  })
})
