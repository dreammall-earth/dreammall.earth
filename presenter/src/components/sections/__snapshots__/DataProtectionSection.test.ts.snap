// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataProtectionSection > renders 4 cards 1`] = `
<div
  class="data-protection-section my-16 v-container"
  data-v-79122f6d=""
>
  <div
    class="v-row section-header"
    data-v-79122f6d=""
  >
    <div
      class="v-col"
      data-v-79122f6d=""
    >
      <h1
        class="section-headline"
        data-v-79122f6d=""
      >
        $t('dataProtection.headline')
      </h1>
    </div>
  </div>
  <div
    class="v-row"
    data-v-79122f6d=""
  >
    <div
      class="v-col"
      data-v-79122f6d=""
    >
      <div
        class="v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated content-card mb-2 pa-4 pa-md-12 h-100"
        data-v-79122f6d=""
      >
        <!---->
        <div
          class="v-card__loader"
        >
          <div
            aria-hidden="true"
            aria-valuemax="100"
            aria-valuemin="0"
            class="v-progress-linear v-theme--light v-locale--is-ltr"
            role="progressbar"
            style="height: 0px; --v-progress-linear-height: 2px;"
          >
            <!---->
            <div
              class="v-progress-linear__background"
              style="width: 100%;"
            />
            <transition-stub
              appear="false"
              css="true"
              name="fade-transition"
              persisted="false"
            >
              <div
                class="v-progress-linear__indeterminate"
              >
                
                <div
                  class="v-progress-linear__indeterminate long"
                />
                <div
                  class="v-progress-linear__indeterminate short"
                />
                
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div
          class="v-card-text"
        >
          
          <h2
            class="card-header"
            data-v-79122f6d=""
          >
            $t('dataProtection.general.headline')
          </h2>
          <p
            class="card-content my-4"
            data-v-79122f6d=""
          >
            $t('dataProtection.general.infoStart')
          </p>
          <ul
            class="card-content mt-2"
            data-v-79122f6d=""
          >
            <li
              data-v-79122f6d=""
            >
              $t('dataProtection.general.company')
            </li>
            <li
              data-v-79122f6d=""
            >
              $t('dataProtection.general.person')
            </li>
            <li
              data-v-79122f6d=""
            >
              $t('dataProtection.general.street')
            </li>
            <li
              data-v-79122f6d=""
            >
              $t('dataProtection.general.zip') $t('dataProtection.general.city')
            </li>
          </ul>
          <p
            class="card-content my-4"
            data-v-79122f6d=""
          >
            $t('dataProtection.general.infoEnd')
          </p>
          
        </div>
        <!---->
        <!---->
        
        <!---->
        <span
          class="v-card__underlay"
        />
        
      </div>
    </div>
    <div
      class="v-col"
      data-v-79122f6d=""
    >
      <div
        class="v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated content-card mb-2 pa-4 pa-md-10 h-100"
        data-v-79122f6d=""
      >
        <!---->
        <div
          class="v-card__loader"
        >
          <div
            aria-hidden="true"
            aria-valuemax="100"
            aria-valuemin="0"
            class="v-progress-linear v-theme--light v-locale--is-ltr"
            role="progressbar"
            style="height: 0px; --v-progress-linear-height: 2px;"
          >
            <!---->
            <div
              class="v-progress-linear__background"
              style="width: 100%;"
            />
            <transition-stub
              appear="false"
              css="true"
              name="fade-transition"
              persisted="false"
            >
              <div
                class="v-progress-linear__indeterminate"
              >
                
                <div
                  class="v-progress-linear__indeterminate long"
                />
                <div
                  class="v-progress-linear__indeterminate short"
                />
                
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div
          class="v-card-text"
        >
          
          <h2
            class="card-header"
            data-v-79122f6d=""
          >
            $t('dataProtection.processing.headline')
          </h2>
          <p
            class="card-content my-4"
            data-v-79122f6d=""
          >
            $t('dataProtection.processing.content')
          </p>
          
        </div>
        <!---->
        <!---->
        
        <!---->
        <span
          class="v-card__underlay"
        />
        
      </div>
    </div>
  </div>
  <div
    class="v-row"
    data-v-79122f6d=""
  >
    <div
      class="v-col"
      data-v-79122f6d=""
    >
      <div
        class="v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated content-card mb-2 pa-4 pa-md-12"
        data-v-79122f6d=""
      >
        <!---->
        <div
          class="v-card__loader"
        >
          <div
            aria-hidden="true"
            aria-valuemax="100"
            aria-valuemin="0"
            class="v-progress-linear v-theme--light v-locale--is-ltr"
            role="progressbar"
            style="height: 0px; --v-progress-linear-height: 2px;"
          >
            <!---->
            <div
              class="v-progress-linear__background"
              style="width: 100%;"
            />
            <transition-stub
              appear="false"
              css="true"
              name="fade-transition"
              persisted="false"
            >
              <div
                class="v-progress-linear__indeterminate"
              >
                
                <div
                  class="v-progress-linear__indeterminate long"
                />
                <div
                  class="v-progress-linear__indeterminate short"
                />
                
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div
          class="v-card-text"
        >
          
          <h2
            class="card-header"
            data-v-79122f6d=""
          >
            $t('dataProtection.revocation.headline')
          </h2>
          <p
            class="card-content mt-2"
            data-v-79122f6d=""
          >
            $t('dataProtection.revocation.content')
          </p>
          
        </div>
        <!---->
        <!---->
        
        <!---->
        <span
          class="v-card__underlay"
        />
        
      </div>
    </div>
  </div>
  <div
    class="v-row"
    data-v-79122f6d=""
  >
    <div
      class="v-col"
      data-v-79122f6d=""
    >
      <div
        class="v-card v-card--flat v-theme--light v-card--density-default v-card--variant-elevated content-card mb-2 pa-4 pa-md-12"
        data-v-79122f6d=""
      >
        <!---->
        <div
          class="v-card__loader"
        >
          <div
            aria-hidden="true"
            aria-valuemax="100"
            aria-valuemin="0"
            class="v-progress-linear v-theme--light v-locale--is-ltr"
            role="progressbar"
            style="height: 0px; --v-progress-linear-height: 2px;"
          >
            <!---->
            <div
              class="v-progress-linear__background"
              style="width: 100%;"
            />
            <transition-stub
              appear="false"
              css="true"
              name="fade-transition"
              persisted="false"
            >
              <div
                class="v-progress-linear__indeterminate"
              >
                
                <div
                  class="v-progress-linear__indeterminate long"
                />
                <div
                  class="v-progress-linear__indeterminate short"
                />
                
              </div>
            </transition-stub>
            <!---->
          </div>
        </div>
        <!---->
        <div
          class="v-card-text"
        >
          
          <h2
            class="card-header"
            data-v-79122f6d=""
          >
            $t('dataProtection.euDispute.headline')
          </h2>
          <p
            class="card-content mt-2"
            data-v-79122f6d=""
          >
            $t('dataProtection.euDispute.content')
          </p>
          
        </div>
        <!---->
        <!---->
        
        <!---->
        <span
          class="v-card__underlay"
        />
        
      </div>
    </div>
  </div>
</div>
`;
